### ISSUE TITLE
All Telegram apps get macOS frozen/hanging on a new MacBook Pro 16-inch M2 Max (2023)

### ISSUE THREAD

**Issue by bountyseek** [overtake/TelegramSwift#1026]
So I just can't use Telegram apps on this new mac. It started  from the beginning after this mac's purchase in April, on previous macOS versions and still continues on new Sonoma 14.0 (and new Telegram versions) so I have to switch to using web browser versions.

My current Telegram version – 10.1.2.252294 Stable.

So basically every 2-3 days (sometimes more often, sometimes less often) when I just do usual scrolling through my contacts in Telegram or chats in folders – just out of nowhere Telegram gets stuck – and the whole MacBook gets frozen, I can't even make a click) – and nothing can be done except reboot using the power button.

Here I recorded a shot clip (blurred due to personal data, but you can still see a cursor) – https://drive.google.com/file/d/1RpMY0sfFJV19C_eYILIiP1KNZo_GxUPj/view 

I tried different Telegram versions (Telegram for macOS, Telegram Desktop) – the same. Old 13.0 macOS, New 14.0 macOS – the same. It's very annoying, Telegram is on of my main working app and I just can't use it( There've  never  been such a problem on a old MacBook Pro 2019 and even oldest MacBook Pro 2013. Maybe some incompatibility with new processor's architecture?

**Comment by hirakujira** [comment]
This also occurs to me (#1017).

**Comment by ysoftware** [comment]
This happens a lot to me as well.

**Comment by TurboKach** [comment]
same for me

macos: Version 13.6 (22G120)
telegram: 10.1.3.252326 Stable

**Comment by smiba** [comment]
Just out of intrest, what tweaks do people have installed? I'm starting to suspect an incompatibility with Bartender 5, so far I've been crash free for almost 24 hours after closing it.

**Comment by hirakujira** [comment]
> Just out of interest, what tweaks do people have installed? I'm starting to suspect an incompatibility with Bartender 5, so far I've been crash free for almost 24 hours after closing it.

No, it's not Bartender I think. I tried to close all apps and scrolled a group chat, it still crashed.

**Comment by smiba** [comment]
> [...] I'm starting to suspect an incompatibility with Bartender 5, so far I've been crash free for almost 24 hours after closing it.

72 hours of no crashes after fully closing Bartender 5, just make sure to actually press "Quit" in the tool itself and not just close it's panel. No clue if I've just been lucky, but I used to experience multiple crashes per day before closing it. 


> I have the same issue with MacBook Pro 14-inch M2 Max (2023) and Sonoma — with absolutely fresh install on clean and first boot.

No tweaks or tools installed at all? Any programs running at the same time?

**Comment by TurboKach** [comment]
i have no tweaks installed. it happens in group chats when i scroll it

**Comment by bountyseek** [comment]
> > [...] I'm starting to suspect an incompatibility with Bartender 5, so far I've been crash free for almost 24 hours after closing it.
> 
> 72 hours of no crashes after fully closing Bartender 5, just make sure to actually press "Quit" in the tool itself and not just close it's panel. No clue if I've just been lucky, but I used to experience multiple crashes per day before closing it.

I have never had Bartender on my macbook, so doubt it's the culprit.
